cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -329,6 +341,14 @@ void decode_page_headers(hostdevice_vector<gpu::ColumnChunkDesc>& chunks,
   chunks.host_to_device(stream);
   gpu::DecodePageHeaders(chunks.device_ptr(), chunks.size(), stream);
   pages.device_to_host(stream, true);
+
+  // validate page encodings (avoiding use of thrust::any_of() NVIDIA/thrust #1016)
+  auto const num_valid_pages = static_cast<size_t>(thrust::count_if(
+    rmm::exec_policy(stream), pages.d_begin(), pages.d_end(), [] __device__(auto const& page) {
+      return is_supported_encoding(page.encoding);
+    }));
+
+  CUDF_EXPECTS(num_valid_pages == pages.size(), "Unsupported page encoding detected");
```

Comment:
Wait, I just see that we already call `pages.device_to_host(stream, true);` in the line right above. That means we can call `is_supported_encoding` on host code:
```suggestion

  // validate page encodings (avoiding use of thrust::any_of() NVIDIA/thrust #1016)
  auto const num_valid_pages = static_cast<size_t>(thrust::count_if(
    rmm::exec_policy(stream), pages.begin(), pages.end(), [] auto const& page) {
      return is_supported_encoding(page.encoding);
    }));

  CUDF_EXPECTS(num_valid_pages == pages.size(), "Unsupported page encoding detected");
```

Reply:
Can it be any_of/all_of now?

Reply:
I was wanting to use the thrust version to do the check in parallel since the number of pages can easily get into the thousands (or higher).  Do you thinks that's not worth worrying about?  Then, yes, I'd switch back to std::any_of/all_of.

Reply:
The runtime of thousands of such trivial checks will be negligible. So you don't need to be worried about it.

Reply:
Ok, it's back on the host.  It's slower by 30us :rofl: (for a totally unscientific single pass of nsys profile).

Reply:
+30us from 1us, or from 1ms? :D

Reply:
30us to 60 us :D